src/randomizer.py

```diff
@@ -1936,6 +1936,52 @@ def visit_Assignment(self, node: ca.Assignment) -> None:
     ast_util.insert_statement(tob, toi, dup)
 
 
+def perm_chain_assignment(
+    fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
+) -> None:
+    """Combine two consecutive assignments into one chain assignment."""
+    cands: List[Tuple(int, ca.Compound)] = []
+
+    def rec(block: Block) -> None:
+        statements = ast_util.get_block_stmts(block, False)
+        for i, (stmt, next_stmt) in enumerate(zip(statements, statements[1:])):
+            if (
+                region.contains_node(stmt)
+                and region.contains_node(next_stmt)
+                and isinstance(stmt, ca.Assignment)
+                and isinstance(next_stmt, ca.Assignment)
+                and (
+                    ast_util.equal_ast(stmt.lvalue, next_stmt.rvalue)
+                    or ast_util.equal_ast(stmt.rvalue, next_stmt.rvalue)
+                )
+                and not ast_util.is_effectful(next_stmt.rvalue)
+            ):
+                cands.append((i, block))
+
+        for stmt in statements:
+            ast_util.for_nested_blocks(stmt, rec)
+
+    rec(fn.body)
+
+    ensure(cands)
+    chosen_assignment_idx, block = random.choice(cands)
+
+    statements = ast_util.get_block_stmts(block, False)
```

Comment:
```suggestion
    statements = ast_util.get_block_stmts(block, True)
```
(doesn't matter but it's semantically right)